### ISSUE TITLE
Can not compile natalie anymore

### ISSUE THREAD

**Issue by rubyFeedback** [natalie-lang/natalie#1717]
I am on ruby 3.3.0 now.

I tried a git clone just now; and run "rake" but it fails:

I am in the directory:

    /Depot/jjjj/natalie/

I then run:

    rake

This is the copy/pasted output I get.

    which: no compiledb in (/usr/bin/:/root/bin/:/opt/bin/:/root/.gem/bin/:/usr/sbin/:/bin/:/sbin/:/usr/local/bin/:/opt/qt/bin/:/usr/local/texlive/2019/bin/x86_64-linux/:/root/.cargo/bin/:/root/.local/bin/:/usr/local/texlive/2023/bin/x86_64-linux)
    git submodule update --init --recursive
    Submodule 'ext/minicoro' (https://github.com/edubart/minicoro.git) registered for path 'ext/minicoro'
    Submodule 'ext/onigmo' (https://github.com/k-takata/Onigmo.git) registered for path 'ext/onigmo'
    Submodule 'ext/prism' (https://github.com/ruby/prism) registered for path 'ext/prism'
    Submodule 'ext/tm' (https://github.com/seven1m/tm) registered for path 'ext/tm'
    Submodule 'ext/zlib' (https://github.com/madler/zlib) registered for path 'ext/zlib'
    Cloning into '/Depot/jjjj/natalie/ext/minicoro'...
    Cloning into '/Depot/jjjj/natalie/ext/onigmo'...
    Cloning into '/Depot/jjjj/natalie/ext/prism'...
    Cloning into '/Depot/jjjj/natalie/ext/tm'...
    Cloning into '/Depot/jjjj/natalie/ext/zlib'...
    Submodule path 'ext/minicoro': checked out '8673ca62ed938c0b436bc2a548f172865f65bf1d'
    Submodule path 'ext/onigmo': checked out '0830382895a303a478de988b7ba7af0f0e86bb6c'
    Submodule path 'ext/prism': checked out '53ef1e8a193cc24d4ce3e0d2d2eade450125ce19'
    Submodule path 'ext/tm': checked out '1e261a54e1b9fa889f189b17b484f0170cac53e8'
    Submodule path 'ext/zlib': checked out '04f42ceca40f73e2978b50e93806c2a18c1281fc'
    bundle check || bundle install
    The following gems are missing
     * minitest-reporters (1.6.1)
     * minitest-parallel_fork (2.0.0)
     * stackprof (0.2.25)
     * rubocop (1.59.0)
     * ansi (1.5.0)
     * ruby-progressbar (1.13.0)
     * language_server-protocol (3.17.0.3)
     * parallel (1.24.0)
     * parser (3.2.2.4)
     * rainbow (3.1.1)
     * regexp_parser (2.8.3)
     * rubocop-ast (1.30.0)
     * unicode-display_width (2.5.0)
     * ast (2.4.2)
    Install missing gems with `bundle install`
    Don't run Bundler as root. Installing your bundle as root will break this application for all non-root users on this machine.
    Bundler 2.5.3 is running, but your lockfile was generated with 2.3.26. Installing Bundler 2.3.26 and restarting using that version.    
    Fetching gem metadata from https://rubygems.org/.    
    Fetching bundler 2.3.26    
    Installing bundler 2.3.26        
    Don't run Bundler as root. Bundler can ask for sudo if it is needed, and installing your bundle as root will break this application for     all non-root users on
    this machine.
    Fetching gem metadata from https://rubygems.org/.........
    Using rake 13.1.0
    Using bundler 2.3.26
    Using minitest 5.20.0
    Fetching ansi 1.5.0
    Using racc 1.7.3
    Using json 2.7.1
    Using rexml 3.2.6
    Fetching ast 2.4.2
    Fetching language_server-protocol 3.17.0.3
    Using builder 3.2.4
    Fetching ruby-progressbar 1.13.0
    Fetching parallel 1.24.0    
    Fetching rainbow 3.1.1    
    Fetching regexp_parser 2.8.3
    Fetching unicode-display_width 2.5.0
    Fetching stackprof 0.2.25
    Fetching minitest-parallel_fork 2.0.0
    Installing ast 2.4.2
    Installing parallel 1.24.0
    Installing ruby-progressbar 1.13.0
    Installing ansi 1.5.0
    Installing unicode-display_width 2.5.0

(And so forth; then:)

Fetching rubocop 1.59.0
Installing rubocop 1.59.0

# Some error occurs:

    /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:93: [BUG] Segmentation fault at 0x0000000000000014
    ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [x86_64-linux]

    -- Control frame information -----------------------------------------------
    c:0024 p:---- s:0174 e:000173 CFUNC  :gets
    c:0023 p:0034 s:0170 e:000169 BLOCK  /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:93
    c:0022 p:0054 s:0163 e:000162 METHOD /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/open3.rb:540
    c:0021 p:0094 s:0152 e:000151 METHOD /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/open3.rb:522
    c:0020 p:0124 s:0141 e:000140 METHOD /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:91
    c:0019 p:0089 s:0125 e:000124 METHOD /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/ext_conf_builder.rb:28
    c:0018 p:0072 s:0109 e:000108 METHOD /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:193
    c:0017 p:0017 s:0098 e:000097 BLOCK  /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:227 [FINISH]
    c:0016 p:---- s:0094 e:000093 CFUNC  :each
    c:0015 p:0084 s:0090 e:000089 METHOD /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:224
    c:0014 p:0016 s:0085 e:000084 METHOD /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/installer.rb:852
    c:0013 p:0034 s:0080 e:000079 METHOD /root/.gem/gems/bundler-2.3.26/lib/bundler/rubygems_gem_installer.rb:72
    c:0012 p:0060 s:0073 e:000072 METHOD /root/.gem/gems/bundler-2.3.26/lib/bundler/rubygems_gem_installer.rb:28
    c:0011 p:0414 s:0069 e:000068 METHOD /root/.gem/gems/bundler-2.3.26/lib/bundler/source/rubygems.rb:207
    c:0010 p:0025 s:0052 e:000051 METHOD /root/.gem/gems/bundler-2.3.26/lib/bundler/installer/gem_installer.rb:54
    c:0009 p:0003 s:0048 e:000047 METHOD /root/.gem/gems/bundler-2.3.26/lib/bundler/installer/gem_installer.rb:16
    c:0008 p:0034 s:0042 e:000041 METHOD /root/.gem/gems/bundler-2.3.26/lib/bundler/installer/parallel_installer.rb:186
    c:0007 p:0007 s:0033 e:000032 BLOCK  /root/.gem/gems/bundler-2.3.26/lib/bundler/installer/parallel_installer.rb:177
    c:0006 p:0009 s:0028 e:000027 METHOD /root/.gem/gems/bundler-2.3.26/lib/bundler/worker.rb:62
    c:0005 p:0030 s:0021 e:000019 BLOCK  /root/.gem/gems/bundler-2.3.26/lib/bundler/worker.rb:57
    c:0004 p:0018 s:0016 e:000015 METHOD <internal:kernel>:187
    c:0003 p:0004 s:0011 e:000010 METHOD /root/.gem/gems/bundler-2.3.26/lib/bundler/worker.rb:54
    c:0002 p:0005 s:0006 e:000005 BLOCK  /root/.gem/gems/bundler-2.3.26/lib/bundler/worker.rb:91 [FINISH]
    c:0001 p:---- s:0003 e:000002 DUMMY  [FINISH]

    -- Ruby level backtrace information ----------------------------------------
    /root/.gem/gems/bundler-2.3.26/lib/bundler/worker.rb:91:in `block (2 levels) in create_threads'
    /root/.gem/gems/bundler-2.3.26/lib/bundler/worker.rb:54:in `process_queue'
    <internal:kernel>:187:in `loop'
    /root/.gem/gems/bundler-2.3.26/lib/bundler/worker.rb:57:in `block in process_queue'
    /root/.gem/gems/bundler-2.3.26/lib/bundler/worker.rb:62:in `apply_func'
    /root/.gem/gems/bundler-2.3.26/lib/bundler/installer/parallel_installer.rb:177:in `block in worker_pool'
    /root/.gem/gems/bundler-2.3.26/lib/bundler/installer/parallel_installer.rb:186:in `do_install'
    /root/.gem/gems/bundler-2.3.26/lib/bundler/installer/gem_installer.rb:16:in `install_from_spec'
    /root/.gem/gems/bundler-2.3.26/lib/bundler/installer/gem_installer.rb:54:in `install'
    /root/.gem/gems/bundler-2.3.26/lib/bundler/source/rubygems.rb:207:in `install'
    /root/.gem/gems/bundler-2.3.26/lib/bundler/rubygems_gem_installer.rb:28:in `install'
    /root/.gem/gems/bundler-2.3.26/lib/bundler/rubygems_gem_installer.rb:72:in `build_extensions'
    /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/installer.rb:852:in `build_extensions'
    /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:224:in `build_extensions'
    /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:224:in `each'
    /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:227:in `block in build_extensions'
    /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:193:in `build_extension'
    /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/ext_conf_builder.rb:28:in `build'
    /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:91:in `run'
    /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/open3.rb:522:in `popen2e'
    /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/open3.rb:540:in `popen_run'
    /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:93:in `block in run'
    /home/Programs/Ruby/3.3.0/lib/ruby/3.3.0/rubygems/ext/builder.rb:93:in `gets'

    -- Threading information ---------------------------------------------------
    Total ractor count: 1
    Ruby thread count for this ractor: 16

    -- Machine register context ------------------------------------------------
     RIP: 0x000056369962b023 RBP: 0x00000000ffffffff RSP: 0x00007f4a874a35b0
     RAX: 0x000056369a436db0 RBX: 0x00007f4a8f0592a0 RCX: 0x0000000000000001
     RDX: 0x0000000000000000 RDI: 0x0000000000000001 RSI: 0x0000000000000001
      R8: 0x0000000000000001  R9: 0x0000000000000000 R10: 0x0000000000000000
     R11: 0x0000000000000246 R12: 0x0000000000000001 R13: 0x00007f4a582e6643
     R14: 0x00007f4a5834bd60 R15: 0x00007f4a5834bd60 EFL: 0x0000000000010286

    -- C level backtrace information -------------------------------------------
    /home/Programs/Ruby/3.3.0/bin/ruby(0x5636997d2c09) [0x5636997d2c09]
    /home/Programs/Ruby/3.3.0/bin/ruby(0x563699969ba8) [0x563699969ba8]
    /home/Programs/Ruby/3.3.0/bin/ruby(0x56369974598d) [0x56369974598d]
    /usr/lib/libc.so.6(0x7f4aad5bdab0) [0x7f4aad5bdab0]
    /home/Programs/Ruby/3.3.0/bin/ruby(0x56369962b023) [0x56369962b023]
    /home/Programs/Ruby/3.3.0/bin/ruby(0x563699668535) [0x563699668535]
    /home/Programs/Ruby/3.3.0/bin/ruby(0x5636996685f0) [0x5636996685f0]

    rake aborted!
    Command failed with status (): [bundle check || bundle install]
    /Depot/jjjj/natalie/Rakefile:614:in `block in <top (required)>'
    /Depot/jjjj/natalie/Rakefile:13:in `block in <top (required)>'
    Tasks: TOP => build_debug => libnatalie => bundle_install
    (See full trace by running task with --trace)

Not sure why it fails. Was bundler always used though? It's weird that bundler
wants to install tons of gems for my current ruby version. I don't understand
that. Is all of that necessary to build natalie now? Very odd. I thought Natalie
is mostly C++, but somehow when running "make" I am told to use "rake",
which then fails dramatically (no surprise - bundler is honestly something
that should have never penalised the ruby ecosystem ... I am surprised
people manage to make bundler actually work).

**Comment by seven1m** [comment]
If you just run `bundle install` yourself, do you get the same segfault?

**Comment by seven1m** [comment]
To answer your question about Bundler, it _should_ only be needed to run the test suite -- not to build Natalie. That's the idea, but I haven't looked into that stuff in a while. I think we can move the `bundle install` bit to be a dependency of `rake test` instead of `rake build`, which would let us bootstrap Natalie without use of Bundler.

**Comment by seven1m** [comment]
@rubyFeedback would you try running `bundle install` on the command line and see what you get?

**Comment by rubyFeedback** [comment]
Hi there Tim,

Sorry for the long delay, I am very unorganized.

I had an unrelated problem - my old fast machine broke down about a week ago or so
due to a faulty power cable. I transitioned to my older machine, which had other issues.

Anyway, to not make this story too long, on that older machine I installed slackware 
15.0 or something like that, then manually upgraded tons of things, compiling
from source (using ruby scripts).

Today I also tested natalie again, using a git clone checkout, and it all compiled
without problem! \o/

I also tested it, it works (although I am a bit confused, in bin/ there is only natalie
binary. This one works fine by the way, I just tested it on my archetypical 
hello_world.rb, but was there not also some bin/irb in the past? Anyway, I think
this is for me to find out, for now I am happy to tell that the issue I have had is
no longer there, I can compile natalie again. I'll have a look at this again in the
future in regards to irb, but other than that I think the issue I reported above is
fixed/gone. \o/
